source/validate.cpp

```diff
@@ -185,6 +185,11 @@ spv_result_t spvValidate(const spv_const_context context,
                                 binary->wordCount, setHeader,
                                 ProcessInstruction, pDiagnostic));
 
+  // Finish analyzing functions, in case a final OpFunctionEnd was not supplied.
+  for (auto& f : vstate.get_functions()) {
+    f.RegisterFunctionEnd();
+  }
```

Comment:
Agreed the spec requires an OpFunctionEnd.  I noticed some tests didn't have it probably because we don't check for it yet?
I'll move it.